src/Symfony/Component/Notifier/Notifier.php

```diff
@@ -89,6 +90,10 @@ private function getChannels(Notification $notification, RecipientInterface $rec
                 throw new LogicException(sprintf('The "%s" channel does not exist.', $channelName));
             }
 
+            if ($channel instanceof SmsChannel && $recipient instanceof NoRecipient) {
```

Comment:
The purpose of the PR

Reply:
Throwing here means that there is no way to try another notifier that might not be a SMS channel, so that's a BC break and a change in the way we manage notifications.